tests/bwc/test_upgrade.py

```diff
@@ -288,3 +290,66 @@ def assert_meta_data(self, version, nodes):
                              cursor.fetchall())
 
             self._process_on_stop()
+
+
+class DefaultTemplateMetaDataCompatibilityTest(NodeProvider, unittest.TestCase):
+    CLUSTER_ID = gen_id()
+
+    CLUSTER_SETTINGS = {
+        'cluster.name': CLUSTER_ID,
+        'es.api.enabled': 'true'
+    }
+
+    SUPPORTED_VERSIONS = (
+        '1.1.x',
+        'latest-nightly',
+    )
+
+    def test_metadata_compatibility(self):
+        nodes = 3
+
+        cluster = self._new_cluster(self.SUPPORTED_VERSIONS[0],
+                                    nodes,
+                                    self.CLUSTER_SETTINGS)
+        cluster.start()
+        with connect(cluster.node().http_url, error_trace=True) as conn:
+            cursor = conn.cursor()
+            cursor.execute("select 1")
+        self._process_on_stop()
+
+        for version in self.SUPPORTED_VERSIONS[1:]:
+            self.assert_dynamic_string_detection(version, nodes)
+
+    def assert_dynamic_string_detection(self, version, nodes):
+        """ Test that a dynamic string column detection works as expected.
+
+        If the cluster was initially created/started with a lower CrateDB
+        version, we must ensure that our default template is also upgraded if
```

Comment:
suggestion: ... is also upgraded, if needed, because it is persisted in the cluster state. That's why re-creating tables would not help.